Clarify the documentation for TLS cipher configuration & defaults

pimterry opened the pull request:
I needed to set `ciphers` on a TLS socket today, and the format of the cipher list string doesn't appear to be documented anywhere.

To work out how it worked, I needed to check the default value, which unfortunately also isn't documented.

This PR fixes both: making it clear that ciphers is a colon-separated list, and documenting the `tls.DEFAULT_CIPHERS` property that is its default value.

jasnell commented:
Yes and no. `tls.DEFAULT_CIPHERS` is set initially to equal `crypto.constants.defaultCoreCipherList` but is mutable once set, whereas `crypto.constants.defaultCoreCipherList` remains a constant.

pimterry commented:
`tls.DEFAULT_CIPHERS` also changes if you use `--tls-cipher-list`, unlike `defaultCoreCipherList`:

```
$ node --tls-cipher-list=aNULL
> require('crypto').constants.defaultCoreCipherList
'TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256:TLS_AES_128_GCM_SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA256:ECDHE-RSA-AES256-SHA384:DHE-RSA-AES256-SHA384:ECDHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA256:HIGH:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!SRP:!CAMELLIA'
> require('tls').DEFAULT_CIPHERS
'aNULL'
```

pimterry commented:
This was approved nearly a year ago and I think it's all good to go. I've just rebased and fixed conflicts, it'd be great to get this merged when somebody has a minute :pray:

pimterry commented:
Closing - a year and a half later https://github.com/nodejs/node/pull/46482/ added an equivalent change, and is now merged, so this is no longer necessary.
